src/lazy/binary/raw/v1_1/value.rs

```diff
@@ -265,9 +268,298 @@ impl<'top> LazyRawBinaryValue_1_1<'top> {
+            return Err(IonError::decoding_error("invalid timestamp length"));
+        }
+
+        let year = (value_bytes[0] as u32) | ((value_bytes[1] & 0x3F) as u32) << 8;
+        let ts_builder = Timestamp::with_year(year);
+        if value_length == 2 {
+            return Ok(RawValueRef::Timestamp(ts_builder.build()?));
+        }
+
+        let month = ((value_bytes[1] & 0xC0) as u32) >> 6 | ((value_bytes[2] & 0x03) as u32) << 2;
+        let day = ((value_bytes[2] & 0x7C) as u32) >> 2;
+        let ts_builder = ts_builder.with_month(month);
+        if value_length == 3 && day == 0 {
+            return Ok(RawValueRef::Timestamp(ts_builder.build()?));
+        }
+
+        let ts_builder = ts_builder.with_day(day);
+        if value_length == 3 {
+            return Ok(RawValueRef::Timestamp(ts_builder.build()?));
+        }
+
+        let hour = ((value_bytes[2] & 0x80) as u32) >> 7 | ((value_bytes[3] & 0x0F) as u32) << 1;
+        let minute = ((value_bytes[3] & 0xF0) as u32) >> 4 | ((value_bytes[4] & 0x03) as u32) << 4;
+        let offset = ((value_bytes[4] & 0xFC) as u32) >> 2 | ((value_bytes[5] & 0x3F) as u32) << 6;
+        let offset = if (offset & 0x800) == 0x800 && offset != 0xFFF {
+            (offset | !0xFFFu32) as i32 // Extend sign for negative 12-bit offsets.
+        } else {
+            offset as i32
+        };
+        let ts_builder = ts_builder.with_hour_and_minute(hour, minute);
+
+        if value_length == 6 {
+            if offset != 0x0FFF {
+                let ts_builder = ts_builder.with_offset(offset);
+                return Ok(RawValueRef::Timestamp(ts_builder.build()?));
+            }
+            return Ok(RawValueRef::Timestamp(ts_builder.build()?));
+        }
+
+        let second = ((value_bytes[5] & 0xC0) as u32) >> 6 | ((value_bytes[6] & 0x0F) as u32) << 2;
+        let ts_builder = ts_builder.with_second(second);
+
+        if value_length == 7 {
+            if offset != 0x0FFF {
+                let ts_builder = ts_builder.with_offset(offset);
+                return Ok(RawValueRef::Timestamp(ts_builder.build()?));
+            }
+            return Ok(RawValueRef::Timestamp(ts_builder.build()?));
+        }
+
+        let scale = FlexUInt::read(&value_bytes[7..], 0)?;
+        let coefficient_start = 7 + scale.size_in_bytes();
+        let coefficient_len = value_length - coefficient_start;
+        let coefficient = FixedUInt::read(&value_bytes[coefficient_start..], coefficient_len, 0)?;
+
+        if scale.value() == 0 {
+            return Err(IonError::decoding_error(
+                "invalid scale for timestamp fractional second",
+            ));
+        }
+
+        let frac_sec = Decimal::new(coefficient, -(scale.value() as i64));
+        if frac_sec.is_greater_than_or_equal_to_one() {
+            return Err(IonError::decoding_error(
+                "invalid fractional second for timestamp",
+            ));
+        }
```

Comment:
`Timestamp` ensures its own invariants for fractional seconds, so these checks are not needed.
```suggestion
        let frac_sec = Decimal::new(coefficient, -(scale.value() as i64));
```